### PR TITLE
[#2568] improvement(trino): Optimize char/varchar mapping for Hive catalog between Gravitino and Trino

### PR DESCRIPTION
**Pull request by yuqi1129**
### What changes were proposed in this pull request?

Changing the char/varchar conversion between Gravitino and Trino.

### Why are the changes needed?

We need to make type conversion more exactly. 

Fix: #2568 

### Does this PR introduce _any_ user-facing change?

N/A.

### How was this patch tested?

UT and IT

### CODE REVIEW

File: trino-connector/src/main/java/com/datastrato/gravitino/trino/connector/catalog/hive/HiveDataTypeTransformer.java [hunk 1]
```diff
@@ -6,28 +6,52 @@
...
-    if (gravitinoType.name() == Name.VARCHAR
-        && ((Types.VarCharType) gravitinoType).length() > HIVE_VARCHAR_MAX_LENGTH) {
-      return Types.VarCharType.of(HIVE_VARCHAR_MAX_LENGTH);
-    }
+    Class<? extends io.trino.spi.type.Type> typeClass = type.getClass();
+    if (typeClass == VarcharType.class) {
+      VarcharType varcharType = (VarcharType) type;
+      if (varcharType.getLength().isEmpty()) {
+        // It was creating a table with column type 'varchar' NOT 'varchar(n)', We do not support
+        // this case
```
Comment (diqiu50) [inline trino-connector/src/main/java/com/datastrato/gravitino/trino/connector/catalog/hive/HiveDataTypeTransformer.java:28]: Why does Hive not support the datatype varchar without specifying a length?  It has conflict with https://trino.io/docs/current/connector/hive.html#basic-usage-examples

Comment (yuqi1129) [reply trino-connector/src/main/java/com/datastrato/gravitino/trino/connector/catalog/hive/HiveDataTypeTransformer.java:28]: Data type varchar without length corresponds to `String` type in Gravitino and there is NO data type `String` actually. 

Please see: 
https://docs.google.com/document/d/1xNkl4Y2t8Bfb8UOOdGBhyQr5DDreAOLmK8FOWiIDiQM/edit

**Comment by yuqi1129** [comment]
@diqiu50 
I have made some modifications back. Please take a look if you have time, thanks.

File: integration-test/src/test/resources/trino-ci-testset/testsets/hive/00007_varchar.sql [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+CREATE TABLE tb05 (id int, name varchar(65535));
+
+SHOW CREATE TABLE "test.gt_hive".varchar_db1.tb05;
+
+CREATE TABLE tb06 (id int, name char);
+
+SHOW CREATE TABLE "test.gt_hive".varchar_db1.tb06;
+
+CREATE TABLE tb07 (id int, name varchar);
+
```
Comment (diqiu50) [inline integration-test/src/test/resources/trino-ci-testset/testsets/hive/00007_varchar.sql:28]: Missing show create table

File: trino-connector/src/main/java/com/datastrato/gravitino/trino/connector/catalog/hive/HiveDataTypeTransformer.java [hunk 3]
```diff
@@ -6,28 +6,48 @@
...
+      if (varcharType.getLength().isEmpty()) {
+        return Types.StringType.get();
+      }
+
+      int length = varcharType.getLength().get();
+      if (length > HIVE_VARCHAR_MAX_LENGTH) {
+        throw new TrinoException(
+            GravitinoErrorCode.GRAVITINO_ILLEGAL_ARGUMENT,
+            "Hive does not support the datatype VARCHAR with the length greater than "
+                + HIVE_VARCHAR_MAX_LENGTH);
```
Comment (diqiu50) [inline trino-connector/src/main/java/com/datastrato/gravitino/trino/connector/catalog/hive/HiveDataTypeTransformer.java:35]: The error message can prompt the user to use varchar as a substitute.

File: integration-test/src/test/resources/trino-ci-testset/testsets/hive/00007_varchar.sql (hunk 2 above)
Comment (yuqi1129) [reply integration-test/src/test/resources/trino-ci-testset/testsets/hive/00007_varchar.sql:28]: OK

**Comment by yuqi1129** [comment]
@diqiu50 
All comments have been resolved.

**Review by diqiu50 — APPROVED** [review]

**Pull request merged**